src/portals/blockchain-frontend/apps/blockchain-frontend/components/credential-container/CredentialMain.tsx

```diff
@@ -0,0 +1,122 @@
+import { useCredentialState } from './state';
+import QRCode from 'qrcode.react';
+
+export default function CredentialMain() {
+  const { formData, previewUrl, setPreviewUrl, handleChange } =
+    useCredentialState();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    try {
+      const response = await submitData(formData);
+      console.log('Form data submitted successfully:', response);
+
+      setPreviewUrl(response.url); // Set preview URL from response
+    } catch (error) {
+      console.error('Error submitting form:', error);
```

Comment:
Show an error to User, A toast preferably

Reply:
Added toast to errors and submissions